README.md

```diff
@@ -6,8 +6,130 @@
+service, you will first need to create a Tiqr server configuration. A brief example is listed below. Note that this
+example is taken from a Symfony parameters file. This YAML is later transformed to a PHP Array. But for readability the
+YAML format is retained in this example:
+
+```yaml
+ Options for the tiqr library
+    tiqr_library_options:
+        general:
+            identifier: 'tiqr.example.com'
+            # The name of the service that uses tiqr for second factor authentication.
+            # Used by both the TiqrService and the legacy Tiqr_Service during enrollment and when sending an
+            # authentication notification.
+            name: 'tiqr'
+            auth_protocol: 'tiqrauth'
+            enroll_protocol: 'tiqrenroll'
+            ocra_suite: 'OCRA-1:HOTP-SHA1-6:QH10-S'
+            logoUrl: '%base_url%/images/header-logo.png'
+            infoUrl: 'https://tiqr.example.com/tiqr_info_url'
+        library:
+          apns:
+            certificate: 'files/apns.pem'
+            environment: production
+        accountblocking:
+            maxAttempts: 5
+            # temporarilyBlockDuration:
+            # maxTemporarilyBlocks:
+        storage:
+            statestorage:
+              type: 'memcache'
+              arguments:
+                prefix: 'tiqr_'
+                servers:
+                  - host: '192.168.66.11'
+            userstorage:
+              type: 'pdo'
+              arguments:
+                table: 'user'
+                dsn: 'mysql:host=192.168.66.11;dbname=tiqr'
+                username: 'tiqr'
+                password: 'super_secret'
+            usersecretstorage:
+              type: 'pdo'
+              arguments:
+                table: 'tiqrusersecret'
+                dsn: 'mysql:host=192.168.66.11;dbname=tiqr-user-secret'
+                username: 'tiqr'
+                password: 'super_secret_2'
+```
+
+### Creation
+Creating the Tiqr_Service is now as simple as creating a new instance with the configuration
+
+```php
+# Convert the Yaml to a PHP array see: https://github.com/OpenConext/Stepup-tiqr/blob/1aa2b686cd4dbffc0d96135042875a881a0e232e/src/Tiqr/TiqrConfiguration.php
+$options = $this->getTiqrConfigurationOptions();
+# Create the Tiqr_Service
+$service = new Tiqr_Service($options)
+```
+
+### Usage
+The service has 22 public methods that are used to enroll a new user, but also to run authentications. The purpose of this
+section is not to be an API documentation. But an example is shown on how the service methods behave.
+
+```php
+# Example for starting an enrollment process, and catching the enrollment key
+$userId = $this->generateId();
+$sessionId = $this->getSessionId();
+$enrollmentKey = $tiqrService->startEnrollmentSession($userId, $this->accountName, $options['general']['identifier']);
+$this->storeEnrollmentKey($enrollmentKey);
+```
+
```

Comment:
This is an incomplete example. The tests contain fully working examples of both enrollment and authentication and also explain what is going on:

- Enrollment: https://github.com/SURFnet/tiqr-server-libphp/blob/develop/library/tiqr/tests/Tiqr_ServiceTest.php#L49
- Authentication via QR code: https://github.com/SURFnet/tiqr-server-libphp/blob/develop/library/tiqr/tests/Tiqr_ServiceTest.php#L126

The soon to be merged TestServer (https://github.com/SURFnet/tiqr-server-libphp/tree/feature/testserver/TestServer) contains a full implementation of enrollment and all three authentications scenarios:
1. authentication via QR
2. authentication via QR of a specific user
3. authentication via push notification of a specific user

Then, of course, there is the Stepup-tiqr implemenation

Reply:
I was very well aware this is an incomple example. It just shows how to interact with the service. Not sure how valuable this is in a readme. Refering to the tests will be far more helpfull to get a comprehensive understanding on the registration and authentication workflows.